### PR TITLE
Feature request for CloudFormation Template

### PR DESCRIPTION
**Pull request by jb897**
Signed-off-by: Josh Bowman <joshua.bowman@sap.com>

### Description

This change allows the template_body property to be accessed from the aws_cloudformation_template resource.
Another AWS SDK API call to get_template is added, along with a function to retrieve the data stored. One caveat, is that the stack_name parameter must be used in the resource, since this is the only parameter that the API call will accept related to the cloudformation_templates

### Issues Resolved
Needed to access information from the template body when accessing a cloudformation_template.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [ ] New functionality includes integration tests/controls
- [ ] New Terraform resources
- [x] Documentation provided or updated for resources 
- [ ] All Integration Tests pass
- [ ] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/aws_cloudformation_template.md [hunk 1]
```diff
@@ -78,6 +78,8 @@ For additional information, see the [AWS documentation on AWS CloudFormation tem
 | resource_types(resource_identifier_summaries) | The resource types of the resource identifier summaries.                                                          |
 | logical_resource_ids                          | The logical resource id of the resource identifier summaries.                                                     |
 | resource_identifiers                          | The resource identifier of the resource identifier summaries.                                                     |
+| template_body                                 | The structure that contains the body of the template 
+(to access this property, the `stack_name` must be passed in as a parameter).   |
```
Comment (soumyo13) [inline docs/resources/aws_cloudformation_template.md:82]:
```suggestion
| template_body                                 | The structure that contains the body of the template (to access this property, the `stack_name` must be passed in as a parameter).   |
```

**Review by soumyo13 — CHANGES_REQUESTED** [review]
Please check the comment.

**Comment by jb897** [comment]
My fault, hadn't updated the branch. Sorry about that.

File: docs/resources/aws_cloudformation_template.md [hunk 2]
```diff
@@ -78,6 +78,7 @@ For additional information, see the [AWS documentation on AWS CloudFormation tem
 | resource_types(resource_identifier_summaries) | The resource types of the resource identifier summaries.                                                          |
 | logical_resource_ids                          | The logical resource id of the resource identifier summaries.                                                     |
 | resource_identifiers                          | The resource identifier of the resource identifier summaries.                                                     |
+| template_body                                 | The structure that contains the body of the template (to access this property, the `stack_name` must be passed in as a parameter).   |
```
Comment (IanMadd) [inline docs/resources/aws_cloudformation_template.md:81]:
```suggestion
| template_body                                 | The structure that contains the body of the template. The `stack_name` must be passed as a parameter to access this property. |
```

Comment (soumyo13) [reply docs/resources/aws_cloudformation_template.md:81]: @jb897 Please check @IanMadd suggestion.

**Review by Nirbhay1997 — COMMENTED** [review]
Hi , @jb897   could you please update the tests with this attribute?

**Comment by jb897** [comment]
> Sorry for the delay. Was out of office due to Thanksgiving.

**Comment by Nirbhay1997** [comment]
@jb897  could you please add unit tests and integration tests for the attribute?

**Comment by sathish-progress** [comment]
> @jb897 could you please add unit tests and integration tests for the attribute?

As requested by @jonlives in the slack channel, to unblock current implementation of the controls by the customer, approving this one!  Tests for this either way is a minor one and can be part of a separate PR ..

Hey @jb897, please raise a separate PR for the tests once you are done with the implementation of them..

**Review by sathish-progress — APPROVED** [review]
LGTM

**Comment by Nirbhay1997** [comment]
Hi  @jb897,  The attribute `template_body`  is not working for me instead  I need to use it like this  
`template_body.template_body`. Could you please parse it properly so that we can use  template_body  ??

**Comment by jb897** [comment]
> Hi @jb897, The attribute `template_body` is not working for me instead I need to use it like this `template_body.template_body`. Could you please parse it properly so that we can use template_body ??

Hi @Nirbhay1997, I hadn't realized I've been adding this locally, do you still want me to make a PR to include this in the resource pack? Apologies for the late response.

**Pull request merged**